lib/services/access_request.go

```diff
@@ -596,13 +596,18 @@ func GetTraitMappings(cms []types.ClaimMapping) types.TraitMappingSet {
 	return types.TraitMappingSet(tm)
 }
 
+// ResourceLister is an interface which can list resources.
+type ResourceLister interface {
+	ListResources(ctx context.Context, req proto.ListResourcesRequest) (*types.ListResourcesResponse, error)
```

Comment:
`ListResources` is defined in a few interfaces in `lib/services`:
- `Presence`
- `RequestValidatorGetter`

We could embed this new interface here.

We also have `api/client.ListResourcesClient` which is exactly this interface. Should we consolidate any or all of these?

Reply:
I am embedding `ResourceLister` into `RequestValidatorGetter` just a few lines below.

I don't think it would make sense to embed it into `Presence` - that's just another layer of indirection for no real benefit IMO.

As for using `api/client.ListResourcesClient` instead or consolidating them - I'm a big proponent of defining interfaces where they are used. Inevitably one of the users is going to need to extend the interface and then they'll be faced with the choice of splitting it up or having the other users require a bigger interface than they really need. I'd rather have a bit of duplication of a 1-line interface than a new dependency.